src/screen/screen.py

```diff
@@ -166,13 +209,27 @@ def manage_display(self) -> None:
             time.sleep(frame_duration)
 
     def start_display_thread(self) -> None:
-        """Creates and starts the thread that manages the display."""
+        """Creates and starts the thread that manages the display.
+        
+        Args:
+            None
+        
+        Returns:
+            None
+        """
         display_thread = threading.Thread(target=self.manage_display)
         display_thread.daemon = True # Exit display thread when main process exits
```

Comment:
I think in this case it's fine since the screen will just display an unfinished frame, but as soon as it starts up again it will update the display correctly. Additionally, this is why I created the SelfReleasingLock, so that if the thread exits before releasing the lock for whatever reason, it will be released in time. But for good practice and best behavior we should consider dealing with it better as you've suggested in the future.

Reply:
I disagree. Any unfinished operations dealing with memory writes are dangerous. There are underlying mechanisms that the OS uses to manage memory. It is for this reason that we halt the device before powering it off. You can take my word for it or if you are really skeptic I can provide you with additional resources to prove the point. It is not fine in this case and it should not be up for consideration. We have to implement this before deploying the MVP.

Reply:
For all threading related stuff, I will create a new branch. Just comment on the code style (not logic), function descriptions, code comments, function names, variable names, etc.